components/script/dom/bindings/structuredclone.rs

```diff
@@ -72,56 +72,36 @@ struct StructuredCloneWriter {
 }
 
 impl StructuredCloneWriter {
-    unsafe fn write_slice(&self, v: &[u8]) {
-        // writers bytesl takes in a vec! and its length
+    unsafe fn write_slice(&self, s: &usize, v: &[u8]) {
+        write_length(self.w, s.to_be());
         let blob_length = v.len();
         assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, blob_length));
     }
-    unsafe fn write_usize(&self, mut u: &usize) {
-        write_length(self.w, u.to_be());
-    }
 }
 
 struct StructuredCloneReader {
     r: *mut JSStructuredCloneReader,
 }
 
 impl StructuredCloneReader {
-    unsafe fn read_bytes(&self) -> usize {
-        // reads and return length
+    unsafe fn read_bytes(&self) -> Vec<u8> {
         let type_length = read_length(self.r);
-        return type_length;
+        let mut v = vec![0u8; type_length];
+        let blob_length = v.len();
+        assert!(JS_ReadBytes(self.r, v.as_mut_ptr() as *mut raw::c_void, blob_length));
```

Comment:
I was confused when I first read this. I would be less confused if we kept using `type_length` everywhere instead of another variable.